### PR TITLE
Define OprExists resolver with enum values

### PR DESCRIPTION
**Pull request by ihadeed**
a more appropriate fix for issue mentioned in #47

### CODE REVIEW

File: examples/apollo/src/SolutionA/makeExecutableSchema.ts [hunk 1]
```diff
@@ -27,6 +27,15 @@ export default function <TContext>(config: IExecutableSchemaDefinition<TContext>
         .join("\n");
 
     const typesSdl = gql(typesSdlRaw);
+    
+    // fetch OprExists as an Enum type
+    const oprExists = typesCache.OprExists as GraphQLEnumType;
```
Comment (yoavkarako) [inline examples/apollo/src/SolutionA/makeExecutableSchema.ts]: This is a problem for all enums, so maybe something along the lines of:
`Object.values(typesCache).filter(isEnumType).map(enumType => ...`

File: examples/apollo/src/SolutionA/makeExecutableSchema.ts [hunk 2]
```diff
@@ -27,6 +27,15 @@ export default function <TContext>(config: IExecutableSchemaDefinition<TContext>
...
+
+    // define a resolver for OprExists to return our custom values instead of enum key
+    config.resolvers['OprExists'] = oprExists.getValues().reduce((resolver, value) => {
```
Comment (yoavkarako) [inline examples/apollo/src/SolutionA/makeExecutableSchema.ts]: the `resolvers` field in the config is nullable

**Comment by ihadeed** [comment]
@yoavkarako thanks for the review & feedback. Just pushed some changes.

---

Out of curiosity, is there a reason why this isn't part of the main package (as a utility function)?
- Is there a case where a user might want to customize those files per use case?
- If so, can we categorize those modifications into specific operations?

I'm new to the GraphQL ecosystem so I'm not very familiar yet with project patterns and requirements.

**Comment by yoavkarako** [comment]
The main package is without any dependencies besides `graphql` itself, and though `graphql-tools` and `apollo` are more or less the standards at this point, I'd rather keep the package as unopinionated as possible.

That being said, it does seem redundant to force users to copy that code section, so I'm thinking of adding a dedicated package for each integration.

**Comment by yoavkarako** [comment]
If you're interested in adding a `prettier` config, please apply it to the entire code base.

**Pull request merged**